core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -204,9 +211,34 @@ public LoggingDecoratorBuilder responseLogLevelMapper(
      * Returns the {@link LogLevel} to use when logging response logs.
      */
     protected final Function<? super RequestLog, LogLevel> responseLogLevelMapper() {
+        if (isExpectedExceptionsSet) {
+            return log -> {
+                final Throwable t = log.responseCause();
+                if (t == null) {
+                    return successfulResponseLogLevel();
+                }
+                final Class<? extends Throwable> clazz = t.getClass();
+                if (expectedExceptions.containsKey(clazz)) {
+                    return expectedExceptions.get(clazz);
+                }
+                return failedResponseLogLevel();
+            };
+        }
         return responseLogLevelMapper;
     }
 
+    /**
+     * Adds an expected exception and the {@link LogLevel} which will be used when the exception occurred.
+     */
+    public LoggingDecoratorBuilder addExpectedException(Class<? extends Throwable> clazz, LogLevel logLevel) {
+        if (isResponseLogLevelMapperSet) {
```

Comment:
I think users might want to use this method and `responseLogLevelMapper` together:
```java
builder.addExpectedException(MyException.class, DEBUG)
       .responseLogLevelMapper(log -> {
            if (infoLog(log.responseHeaders())) {
                // Use INFO depending on the context of the headers.
                return INFO;
            }
            ...
       });
```
In order to make it work, we need to merge #3972 first and keep working on it. 😅

Reply:
I will check this PR.
After PR merged, I redesign this feature.

Reply:
@minwoox 
I fixed it so that both `addExpectedException ` and `responseLogLevelMapper` can be used at the same time.
Please review again.